lib/storage/src/content_manager/snapshots/recover.rs

```diff
@@ -2,20 +2,54 @@ use std::path::Path;
 
 use collection::collection::Collection;
 use collection::config::CollectionConfig;
-use collection::operations::snapshot_ops::SnapshotRecover;
+use collection::operations::snapshot_ops::{SnapshotPriority, SnapshotRecover};
 use collection::shards::replica_set::ReplicaState;
+use collection::shards::shard::{PeerId, ShardId};
 use collection::shards::shard_config::ShardType;
 use collection::shards::shard_versioning::latest_shard_paths;
 
 use crate::content_manager::snapshots::download::{download_snapshot, downloaded_snapshots_dir};
 use crate::{StorageError, TableOfContent};
 
+async fn activate_shard(
```

Comment:
We can rename this method to show that it is no-replica case. And remove comment

Reply:
Function should not assume the context where it was called.